android/beagle/src/test/java/br/com/zup/beagle/android/context/ContextDataManagerTest.kt

```diff
@@ -366,4 +363,31 @@ class ContextDataManagerTest : BaseTest() {
         // When
         contextDataManager.evaluateContexts()
     }
+
+    @Test
+    fun init_must_add_observer_to_GlobalContext_and_validate_the_updateGlobalContext_method(){
+        // Given
+        val globalContextObserver = slot<GlobalContextObserver>()
+        val contextData = ContextData("global","")
+        val globalContextMock = mockk<ContextBinding>(relaxed = true){
+            every { copy(any(),any(),any()) } returns this
+        }
+        val cache = mockk<LruCache<String, Any>>(relaxed = true)
+        every { globalContextMock.cache } returns cache
+        every { GlobalContext.observeGlobalContextChange(capture(globalContextObserver)) } just Runs
+
+        // When
+        val contextDataManager = spyk<ContextDataManager>(recordPrivateCalls = true)
+        val contexts = contextDataManager.getPrivateField<MutableMap<Int, ContextBinding>>("contexts")
+        contextDataManager.setPrivateField("globalContext", globalContextMock)
+        globalContextObserver.captured.invoke(contextData)
+
+        // Then
+        verify{
```

Comment:
What do you think of using `verifyOrder`? Because in your method the order in which you invoke the methods makes a difference, consequently this should be something to be tested.
```suggestion
        verifyOrder{
```

Reply:
change done!